### ISSUE TITLE
实现 Assistant 库以降低犯错几率

### ISSUE THREAD

**Issue by xxr3376** [MegEngine/MegEngine#52]
背景介绍：
- 像 BN.momentum / optimizer.momentum 这类在业界目前有一定歧义的定义，未来应该还会遇到很多，我们应该做一些假设来让帮助用户少犯错，降低排查成本

其他选择：
- 直接 assert 或者抛 exception 肯定是不合适的，我们并不能预料到全部的用法，万一用户真的就要这么做就不好了
- 用 logger 打 warning 太烦了，还不能 ignore
- 用 python 原本的 warnings 库：功能上都可以，但是使用体验非常诡异

希望达到玩游戏的时候的那种辅助提示的效果
比如 momentum < 0.5 的时候，提示："The momentum of batch normalization layer rarely uses a value less than 0.5, Please check the document for momentum's definition, which is different from PyTorch."
至多提示一次，如果用户觉得我明白风险，可以显式用 API 表明”我了解这个风险”
